pkg/utils/report/results.go

```diff
@@ -156,7 +156,7 @@ func ToPolicyReportResult(policyType engineapi.PolicyType, policyName string, ru
 
 func EngineResponseToReportResults(response engineapi.EngineResponse) []policyreportv1alpha2.PolicyReportResult {
 	pol := response.Policy()
-	policyName := pol.GetName()
+	policyName, _ := cache.MetaNamespaceKeyFunc(pol.AsKyvernoPolicy())
```

Comment:
@coder12git - any reason why you reverted this?

Reply:
yes, because i have added if and else statement inside ToPolicyReport function, so if it will be VPA, in that case policy name will be extracted from rule response.

Reply:
```
else {
		result = policyreportv1alpha2.PolicyReportResult{
			Source:  "ValidatingAdmissionPolicy",
			Policy:  ruleResult.Name(),
			Message: ruleResult.Message(),
			Result:  toPolicyResult(ruleResult.Status()),
			Timestamp: metav1.Timestamp{
				Seconds: time.Now().Unix(),
			},
		}
		if ruleResult.ValidatingAdmissionPolicyBinding() != nil {
			addProperty("binding", ruleResult.ValidatingAdmissionPolicyBinding().Name, &result)
		}
	}
```

Reply:
Its best to replace type based if-else statements, by delegating to the type itself.

I will take a look and update.